controllers/consumer_controller.go

```diff
@@ -324,33 +307,19 @@ func (r *ConsumerReconciler) constructDeployment(consumer konsumeratorv1alpha1.C
 		},
 		Spec: consumer.Spec.DeploymentTemplate,
 	}
-	deploy.Annotations[managedPartitionAnnotation] = strconv.Itoa(int(partition))
-	deploy.Annotations[deploymentGeneration] = strconv.Itoa(int(*consumer.Status.ObservedGeneration))
-	var partitionKey string
-	if consumer.Spec.PartitionEnvKey != "" {
-		partitionKey = consumer.Spec.PartitionEnvKey
-	} else {
-		partitionKey = defaultPartitionEnvKey
-	}
-	for containerIndex := range deploy.Spec.Template.Spec.Containers {
-		name := deploy.Spec.Template.Spec.Containers[containerIndex].Name
-		predictor := predictors.NewNaivePredictor(r.Log, store, consumer.Spec.Autoscaler.Prometheus)
-		limits := predictors.GetResourcePolicy(name, &consumer.Spec)
-		resources := predictor.Estimate(name, limits, partition)
-		deploy.Spec.Template.Spec.Containers[containerIndex].Resources = *resources
-		envs := deploy.Spec.Template.Spec.Containers[containerIndex].Env
-		envs = setEnvVariable(envs, partitionKey, strconv.Itoa(int(partition)))
-		envs = setEnvVariable(envs, gomaxprocsEnvKey, goMaxProcsFromResource(resources.Limits.Cpu()))
-		deploy.Spec.Template.Spec.Containers[containerIndex].Env = envs
-	}
-	if err := ctrl.SetControllerReference(&consumer, deploy, r.Scheme); err != nil {
-		return nil, err
-	}
-	return deploy, nil
+	deploy.Annotations[partitionAnnotation] = strconv.Itoa(int(partition))
+	deploy.Annotations[generationAnnotation] = strconv.Itoa(int(*consumer.Status.ObservedGeneration))
+	return deploy
+}
+
+func estimateResources(predictor *predictors.NaivePredictor, containerName string, consumerSpec *konsumeratorv1alpha1.ConsumerSpec, partition int32) corev1.ResourceRequirements {
```

Comment:
The type of first argument should be an interface with `Estimate(...)` method. This will make it easier to later plug in different predictors.